src/main/scala/apiimport/persistence/ManifestPersistor.scala

```diff
@@ -0,0 +1,130 @@
+package apiimport.persistence
+
+import java.sql.Timestamp
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import apiimport.Db
+import apiimport.manifests.VoyageManifestParser.VoyageManifest
+import apiimport.slickdb.VoyageManifestPassengerInfoTable
+import drtlib.SDate
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.matching.Regex
+import scala.util.{Failure, Success, Try}
+
+
+case class ManifestPersistor(db: Db)(implicit ec: ExecutionContext) {
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  val dqRegex: Regex = "drt_dq_([0-9]{2})([0-9]{2})([0-9]{2})_[0-9]{6}_[0-9]{4}\\.zip".r
+
+  val manifestTable = VoyageManifestPassengerInfoTable(db.tables)
+
+  import db.tables.profile.api._
+
+  def zipFileDate(fileName: String): Option[SDate] = fileName match {
+    case dqRegex(year, month, day) => Option(SDate(s"20$year-$month-$day"))
+    case _ => None
+  }
+
+  val oneDayMillis: Long = 60 * 60 * 24 * 1000L
+
+  def addPersistence(zipTries: Source[(String, Try[List[(String, Try[VoyageManifest])]]), NotUsed]): Source[Int, NotUsed] = zipTries
```

Comment:
this is a slightly confusing function name, should it just be `persist`?

Reply:
It's a deliberate name as it adds persistence to the stream, rather than actually doing the persisting directly
Maybe addPersistenceToStream instead?